txpipe/twopoint_null_tests.py

```diff
@@ -766,3 +767,272 @@ def write_output(self, source_list, lens_list, meta, results):
+            B-mode method of Becker and Rozo 2015 http://arxiv.org/abs/1412.3851
+            '''
+            print('--------------hybrideb-----------')
+
+            # Check if nbins is less than 1000, and throw warning.
+            if self.config['Ntheta']<1000:
+                print("WARNING: Calculating hybridEB using Ntheta<1000, which may lead to inaccurate results.")
+
+            # Computing the weights takes a few minutes so its a lot faster
+            # to precompute and load them in subsequent runs.
+            file_precomputed_weights = self.get_output("twopoint_data_fourier_pureB")+ f".precomputedweights.pkl"
+
+            if os.path.exists(file_precomputed_weights):
+                print(f"{self.rank} WARNING: Using precomputed weights from revious run: {file_precomputed_weights}")
+                with open(file_precomputed_weights, "rb") as f:
+                    geb_dict = pickle.load(f)
+            
+            else:
+                heb = hybrideb.HybridEB(self.config['theta_min'], self.config['theta_max'], self.config['Ntheta'])
+                beb = hybrideb.BinEB(self.config['theta_min'], self.config['theta_max'], self.config['Ntheta'])
+                geb = hybrideb.GaussEB(beb, heb)
+                geb_dict = {}
+                for i in range(0,self.config['Nell']):
+                    geb_dict['%d'%(i+1)] = {}
+                    for j in range(0,6):
+                        geb_dict['%d'%(i+1)]['%d'%(j+1)]=geb(i)[j]
+            
+                with open(file_precomputed_weights, 'wb') as handle:
+                    pickle.dump(geb_dict, handle, protocol=pickle.HIGHEST_PROTOCOL)
+
+            En0  = np.zeros(self.config["Nell"])
+            Bn0  = np.zeros(self.config["Nell"])
+            En   = np.zeros((self.config["Nell"],self.config["Nsims"]))
+            Bn   = np.zeros((self.config["Nell"],self.config["Nsims"]))
+            ell  = np.zeros(self.config["Nell"])
+
+            filename = self.get_input("twopoint_data_real_raw")
+            s    = sacc.Sacc.load_fits(filename)
+
+            Qxip = sacc.standard_types.galaxy_shear_xi_plus
+
+            source_tracers = set()
+            for b1, b2 in s.get_tracer_combinations(Qxip):
+                source_tracers.add(b1)
+                source_tracers.add(b2)
+
+            nbin_source = max(len(source_tracers), 1)
+
+            corvarr = np.zeros( (int(nbin_source*(nbin_source+1)/2*self.config["Nell"]), self.config["Nsims"])  )
+
+            ret=np.zeros((nbin_source,nbin_source,self.config["Nell"]))
+
+            c = 0
+            for zi in range(0,nbin_source):
+                for zj in range(zi,nbin_source):
+                    ii = self.config["Nell"]*c
+                    ff = self.config["Nell"]*(c+1)
+                    
+                    # Need all 4 block of the covariance
+                    filename = self.get_input("twopoint_data_real_raw")
+                    tmp      = sacc.Sacc.load_fits(filename)
```

Comment:
There's no need to read the sacc file in every time, you can read it once outside the loop and then `tmp.clone()` in the loop.

Reply:
Fixed.

Reply:
Actually I'm finding that sacc.clone() doesn't exist, but .copy() does so I've switched to that.